R/loopRuns.R

```diff
@@ -36,59 +36,66 @@ loopRuns <- function(mydir, user = NULL, colors = TRUE, sortbytime = TRUE) {
   if (file.exists("/p")) {
     coltitles <- c(paste0("Folder", paste(rep(" ", len - 6), collapse = "")),
       "Runtime    ", "inSlurm ", "RunType    ", "RunStatus        ", "Iter            ",
-      "Conv                 ", "modelstat          ", "Mif     ", "inAppResults")
-    lenCols <- c(nchar(coltitles)[-length(coltitles)], 5)
+      "Conv                 ", "modelstat          ", "Mif", "AppResults")
+    lenCols <- c(nchar(coltitles)[-length(coltitles)], 3)
   } else {
     coltitles <- c(paste0("Folder", paste(rep(" ", len - 6), collapse = "")),
-      "Runtime    ", "RunType    ", "RunStatus        ", "Mif     ",
+      "Runtime    ", "RunType    ", "RunStatus        ", "Mif",
       "Conv                 ", "Iter            ", "modelstat          ")
     lenCols <- nchar(coltitles)
   }
+  if (colors) {
+    cat("# Color code: ", yellow("pending"), ", ", cyan("running"), ", ", underline(green("converged")), ", ",
+        blue("converged with INFES"), ", ", green("finished"), ", ", red("error"), "\n", sep = "")
+  }
   message(paste(coltitles, collapse = colSep))
-
   for (i in mydir) {
 
     if (!file.exists(paste0(i, "/", grep("^config.*", dir(i), value = TRUE)[1]))) next # do not report on folders that do not contain runs
```

Comment:
- save the run status as `status` and use that to identify colors instead of grepping around in the long `out` string where it is difficult to avoid matching the wrong column
- print in black (or white, depending on background) if none of the `if` matches